### PR TITLE
Remove usages of Guice

### PR DESCRIPTION
**Pull request by basil**
Replacing deprecated usages of Guice with non-deprecated non-Guice equivalents.

### CODE REVIEW

**Comment by basil** [comment]
[BOM testing passed](https://ci.jenkins.io/job/Tools/job/bom/job/PR-783/1/)

**Review by timja — APPROVED** [review]

File: src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java [hunk 1]
```diff
@@ -287,22 +295,21 @@ private void postSettlement() {
...
 
     /**
      * Checks if the given user can settle this input.
      */
-    private boolean canSettle(Authentication a) {
-        String submitter = input.getSubmitter();
+    private static boolean canSettle(Run<?, ?> run, String submitter, Authentication a) {
         if (submitter==null)
-            return getRun().getParent().hasPermission(Job.BUILD);
+            return run.getParent().hasPermission(Job.BUILD);
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java:312]: https://github.com/jenkinsci/acceptance-test-harness/issues/727

**Review by jglick — CHANGES_REQUESTED** [review]

File: src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStep.java [hunk 2]
```diff
@@ -31,7 +36,7 @@
  *
  * @author Kohsuke Kawaguchi
  */
-public class InputStep extends AbstractStepImpl implements Serializable {
+public class InputStep extends Step implements Serializable {
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStep.java:39]: Also not compatible in this case.

Comment (jglick) [reply src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStep.java:39]: (Part of `program.dat`. Also see https://github.com/jenkinsci/pipeline-plugin/commit/7a0f10c21ddc40dc20f62d1b9d27633a7bf8aed4)

**Comment by basil** [comment]
Closing in favor of #69.

**Pull request closed without merging**